### PR TITLE
Jwt token

### PR DESCRIPTION
**Pull request by Fatimata-dev**
Use of the token to manage a user's connection and protect routes depending on the state of the user connected or not

### CODE REVIEW

File: src/app/components/add-expedition/add-expedition.component.ts [hunk 1]
```diff
@@ -17,8 +18,8 @@ export class AddExpeditionComponent {
   isExpeditionModalOpen : boolean = false;
   title : string = "";
   authService:AuthenticationService = inject(AuthenticationService);
-  user = this.authService.getUser();
-
+  //user = this.authService.getUser();
```
Comment (LucieLDa) [inline src/app/components/add-expedition/add-expedition.component.ts]: dead code

File: src/app/components/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class SidebarComponent implements OnInit{
   }
 
   ngOnInit(): void {
-    this.user = this.authService.getUser();
+    //this.user = this.authService.getUser();
```
Comment (LucieLDa) [inline src/app/components/sidebar/sidebar.component.ts]: dead code

File: src/assets/pictures/clock-icon.png
Comment (LucieLDa) [inline src/assets/pictures/clock-icon.png:1]: Clock-icon is not longer used in this project, it was replaced by a font awesome icon

**Review by LucieLDa — CHANGES_REQUESTED** [review]

**Review by LucieLDa — APPROVED** [review]

File: src/app/components/add-expedition/add-expedition.component.ts [hunk 3]
```diff
@@ -16,11 +17,17 @@ import { AuthenticationService } from '../../service/User/authentication.service
 export class AddExpeditionComponent {
   isExpeditionModalOpen : boolean = false;
   title : string = "";
-  authService:AuthenticationService = inject(AuthenticationService);
-  user = this.authService.getUser();
+  userService:UserService = inject(UserService);
+  tokenService:TokenService = inject(TokenService);
+  user!:User;
 
+  userInfo:any ;
```
Comment (Benja-san) [inline src/app/components/add-expedition/add-expedition.component.ts]: if id only id is needed, use number type and retrieve the id directly

File: src/app/app.routes.ts [hunk 4]
```diff
@@ -43,8 +45,9 @@ export const routes: Routes = [
    },
 
   {
-    path:'profil/:id',
+    path:'profile/:id',
```
Comment (Benja-san) [inline src/app/app.routes.ts:48]: keep profil for french content

File: src/app/components/add-expedition/add-expedition.component.ts [hunk 5]
```diff
@@ -16,11 +17,17 @@ import { AuthenticationService } from '../../service/User/authentication.service
 export class AddExpeditionComponent {
   isExpeditionModalOpen : boolean = false;
   title : string = "";
-  authService:AuthenticationService = inject(AuthenticationService);
-  user = this.authService.getUser();
+  userService:UserService = inject(UserService);
+  tokenService:TokenService = inject(TokenService);
+  user!:User;
```
Comment (Benja-san) [inline src/app/components/add-expedition/add-expedition.component.ts:22]: are you sure to retrieve a user ?

File: src/app/components/nav-bar/nav-bar.component.ts [hunk 6]
```diff
@@ -14,16 +15,14 @@ import { AuthenticationService } from '../../service/User/authentication.service
...
 export class NavBarComponent { 
   authService:AuthenticationService = inject(AuthenticationService)
+  tokenService:TokenService = inject(TokenService)
   public isMobileView!: boolean;
   public innerWidth: any;
   router:Router = inject(Router);
-  user?:User
-  id=0;
-  isLogged =false;
+  userInfo: any = {};
```
Comment (Benja-san) [inline src/app/components/nav-bar/nav-bar.component.ts]: same here

File: src/app/components/sidebar/sidebar.component.ts [hunk 7]
```diff
@@ -24,7 +26,9 @@ export class SidebarComponent implements OnInit{
   screenWidth = 0 ;
   navData=sidebarData;
   user?:User;
-  authService:AuthenticationService = inject(AuthenticationService);
+  userInfo:any;
```
Comment (Benja-san) [inline src/app/components/sidebar/sidebar.component.ts]: same here

File: src/app/pages/login/login.component.ts [hunk 8]
```diff
@@ -18,7 +18,9 @@ export class LoginComponent {
   isPasswordHidden = true;
   errorMsg='';
   loginError = false;
+  token: any = {};
```
Comment (Benja-san) [inline src/app/pages/login/login.component.ts]: use type or model for token

File: src/app/pages/task-page/task-page.component.ts [hunk 9]
```diff
@@ -46,14 +47,13 @@ export class TaskPageComponent {
...
-    if(!this.user) return;
-    this.userService.getUserExpeditions(this.user.id).subscribe((expeditions) => {
+    this.userService.getUserExpeditions(this.userInfo.id).subscribe((expeditions) => {
       this.expeditionList = expeditions;
     });
   }
   
   onReceiveNewExpedition(event : Expedition){
-      this.expeditionService.addExpedition(event, this.user?.id || 0).subscribe((ex) => {
+      this.expeditionService.addExpedition(event, this.userInfo.id || 0).subscribe((ex) => {
```
Comment (Benja-san) [inline src/app/pages/task-page/task-page.component.ts]: why or 0 ?

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**